core/src/SYCL/Kokkos_SYCL_Half_MathematicalFunctions.hpp

```diff
@@ -0,0 +1,190 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#ifndef KOKKOS_SYCL_HALF_MATHEMATICAL_FUNCTIONS_HPP_
+#define KOKKOS_SYCL_HALF_MATHEMATICAL_FUNCTIONS_HPP_
+
+namespace Kokkos {
+#ifdef KOKKOS_IMPL_SYCL_HALF_TYPE_DEFINED
+
+#define KOKKOS_SYCL_HALF_UNARY_FUNCTION(OP)              \
```

Comment:
Same comment as in CUDA about doing it at one place only:
```C++
#define KOKKOS_IMPL_SYCL_HALF_UNARY_FUNCTION_FUNC_TYPE(FUNC, TYPE) ...

#if defined(KOKKOS_IMPL_SYCL_HALF_TYPE_DEFINED) && defined(KOKKOS_IMPL_SYCL_BHALF_TYPE_DEFINED)
#define KOKKOS_IMPL_SYCL_HALF_UNARY_FUNCTION(FUNC) \
  KOKKOS_IMPL_SYCL_HALF_UNARY_FUNCTION_FUNC_TYPE(FUNC, half_t) \
  KOKKOS_IMPL_SYCL_HALF_UNARY_FUNCTION_FUNC_TYPE(FUNC, bhalf_t)
#elif defined(KOKKOS_IMPL_SYCL_HALF_TYPE_DEFINED)
#define KOKKOS_IMPL_SYCL_HALF_UNARY_FUNCTION(FUNC) \
  KOKKOS_IMPL_SYCL_HALF_UNARY_FUNCTION_FUNC_TYPE(FUNC, half_t) \
#elif ...

KOKKOS_IMPL_SYCL_HALF_UNARY_FUNCTION(foo)
KOKKOS_IMPL_SYCL_HALF_UNARY_FUNCTION(bar)
```
You get the idea

Reply:
There are much fewer supported functions for `bhalf_t` than for `half_t` for `SYCL` in this pull request.

Reply:
Then I guess ignore my suggestion